### PR TITLE
Microsoft graph security filter office365 doesn't work

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-19504

## Description
The filter parameter in msg-search-alerts is not supported for alerts of provider office 365 due to API limitations.
For more info see: https://github.com/microsoftgraph/security-api-solutions/issues/56

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 1]
```diff
@@ -44,24 +44,15 @@ def search_alerts(self, last_modified, severity, category, vendor, time_from, ti
         if last_modified:
             filters.append("lastModifiedDateTime gt {}".format(get_timestamp(last_modified)))
         if category:
-            filters.append("contains(category, '{}')".format(category))
+            filters.append("category eq '{}'".format(category))
```
Comment (MosheEichler) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:47]: Revert

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 2]
```diff
@@ -44,24 +44,15 @@ def search_alerts(self, last_modified, severity, category, vendor, time_from, ti
...
-            # We use the `contains` function instead of the `eq` operator (as a workaround), For example:
-            # "category eq 'Malware' and severity eq 'High'" => "contains(category, 'Malware') and severity eq 'High'".
-            # For more info please see: https://learn.microsoft.com/en-us/graph/filter-query-parameter.
-            filter_list = [f.strip() for f in filter_query.lower().split('and')]
-            for filter_item in filter_list:
-                if 'category eq' in filter_item:
-                    category = filter_item.split()[-1]
-                    filter_item = "contains(category, {})".format(category)
-                filters.append(filter_item)
+            filters.append("{}".format(filter_query))
```
Comment (MosheEichler) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:55]: Revert

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity_test.py [hunk 3]
```diff
@@ -260,22 +260,3 @@ def test_fetch_incidents_command(mocker):
...
-    ("Category eq 'Malware'", "contains(category, 'malware')")
-])
-def test_filter_query(filter_query, expected_filter_query, mocker):
-    from MicrosoftGraphSecurity import MicrosoftClient
-    mocker.patch.object(MicrosoftClient, 'http_request', side_effect=mock_request)
-
-    response = client_mocker.search_alerts(last_modified=None, severity=None, category=None, vendor=None,
-                                           time_from=None, time_to=None, filter_query=filter_query)
-
-    assert response.get('$filter') == expected_filter_query
```
Comment (MosheEichler) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity_test.py:281]: Revert

**Review by MosheEichler — COMMENTED** [review]
Revert some changes to the previews version
https://github.com/demisto/content/pull/23148

**Comment by ShirleyDenkberg** [comment]
@gal-forer Doc review completed.

**Review by gal-forer — APPROVED** [review]

**Pull request merged**